### DISCUSSION TITLE
Why can I customize some style properties, but not others? [v1]

### DISCUSSION THREAD

**Discussion by vonwao** [chakra-ui/chakra-ui#2452] [Q&A] [answered]
I'm trying to customize my UI elements (specifically Button and Select, at this time), and I noticed that some props work, but others don't.  

 Button: {
    // 1. We can update the base styles
    baseStyle: {
      bg: "green.200",  // DOESN'T WORK
      color: "blue.500",
      borderRadius: "full",
      borderWidth: 1,
      borderColor: "blue.500",
    },
  },
  Select: {
    baseStyle: {  // None of these work
      bg: "green.200",
      color: "blue.500",
      borderRadius: "full",
      borderWidth: 1,
      borderColor: "blue.500",
    },
  },

Can anyone explain the reason or give insights on this?

**Comment by chasinhues** [comment, marked as answer]
When customising component styles in the theme, keep in mind that: 
- some components have `variants`
- [multipart (AKA composite components)](https://next.chakra-ui.com/docs/theming/advanced#multipart-or-composite-component) have `parts` as well. 

#### Button
Any customisations you make to the `baseStyle` may also impact the different [variants](https://next.chakra-ui.com/docs/form/button#button-variants). In the case of `Button`, it might be better to target specific variants that you want to apply the customisations for some styles. 

```jsx
Button: {
  baseStyle: {
    borderRadius: "full"
  },
  variants: {
    solid: {
      bg: "green.200",
      color: "blue.500",
      borderWidth: 1,
      borderColor: "blue.500",
    }
  }
}
```

#### Select
If you check out the [source code for `Select`](https://github.com/chakra-ui/chakra-ui/blob/793f1dd6cf62985a5b6391ac2e843aeaf9ee6fac/packages/theme/src/components/select.ts#L6), you'll see that it has a property called `parts`:
```jsx
...
const parts = ["field", "icon"]
...
```
This indicates that it is a multipart component. These components have a [slightly different approach](https://next.chakra-ui.com/docs/theming/component-style#styling-multipart-components) to customising them, in that you have to target each part inside `baseStyle`/`variants`/etc.. 

> Also note from reviewing the source code that `Select` [imports its `sizes`, `defaultProps`, and `variants` from the `Input` component](https://github.com/chakra-ui/chakra-ui/blob/793f1dd6cf62985a5b6391ac2e843aeaf9ee6fac/packages/theme/src/components/select.ts#L4), so it has the same `variants` as `Input`.

```jsx
Select: {
  baseStyle: {
    field: {
      // base styles for the field go here
    },
    icon: {
      // base styles for the icon go here
    }
  },
  variants: {
    solid: {
      field: {
        // custom styles for the solid variant of the field part of Select go here
      },
      icon: {
        // custom styles for the solid variant of the icon part of Select go here
      }
    }
  }
}
```

**Reply by vonwao** [reply]
@ChasinHues  Good info, thanks!   Do you mind if I contribute this info to the docs?   Or would you like to do that youreself?   I added something else last week, and getting into contributing some improvements to docs.

**Reply by chasinhues** [reply]
Please go for it, all yours :)

**Reply by seloner** [reply]
> When customising component styles in the theme, keep in mind that:
> 
> * some components have `variants`
> * [multipart (AKA composite components)](https://next.chakra-ui.com/docs/theming/advanced#multipart-or-composite-component) have `parts` as well.
> 
> #### Button
> Any customisations you make to the `baseStyle` may also impact the different [variants](https://next.chakra-ui.com/docs/form/button#button-variants). In the case of `Button`, it might be better to target specific variants that you want to apply the customisations for some styles.
> 
> ```js-jsx
> Button: {
>   baseStyle: {
>     borderRadius: "full"
>   },
>   variants: {
>     solid: {
>       bg: "green.200",
>       color: "blue.500",
>       borderWidth: 1,
>       borderColor: "blue.500",
>     }
>   }
> }
> ```
> 
> #### Select
> If you check out the [source code for `Select`](https://github.com/chakra-ui/chakra-ui/blob/793f1dd6cf62985a5b6391ac2e843aeaf9ee6fac/packages/theme/src/components/select.ts#L6), you'll see that it has a property called `parts`:
> 
> ```js-jsx
> ...
> const parts = ["field", "icon"]
> ...
> ```
> 
> This indicates that it is a multipart component. These components have a [slightly different approach](https://next.chakra-ui.com/docs/theming/component-style#styling-multipart-components) to customising them, in that you have to target each part inside `baseStyle`/`variants`/etc..
> 
> > Also note from reviewing the source code that `Select` [imports its `sizes`, `defaultProps`, and `variants` from the `Input` component](https://github.com/chakra-ui/chakra-ui/blob/793f1dd6cf62985a5b6391ac2e843aeaf9ee6fac/packages/theme/src/components/select.ts#L4), so it has the same `variants` as `Input`.
> 
> ```js-jsx
> Select: {
>   baseStyle: {
>     field: {
>       // base styles for the field go here
>     },
>     icon: {
>       // base styles for the icon go here
>     }
>   },
>   variants: {
>     solid: {
>       field: {
>         // custom styles for the solid variant of the field part of Select go here
>       },
>       icon: {
>         // custom styles for the solid variant of the icon part of Select go here
>       }
>     }
>   }
> }
> ```

But lets say you want to change the border radius for the Input Component . Should I target the field baseStyle or the sizes ? If I target field baseStyle I see no change . Then when I target sizes I need to change all sizes border radius  if I want to use different sizes in the app .

**Reply by chasinhues** [reply]
> But lets say you want to change the border radius for the Input Component . Should I target the field baseStyle or the sizes ? If I target field baseStyle I see no change . Then when I target sizes I need to change all sizes border radius if I want to use different sizes in the app .

Ah as you mentioned it doesn't seem like customising `borderRadius` in `baseStyles` works for `Input`. Yeah that's a tricky one. 

You can either add the border radios customisation to the sizes, or to the variants. I would suggest you add the customisation to whichever one you're using less of (less effort). E.g. if you're using only one variant but all three sizes, then if you just add the customisation to the variant you're using, you only need to make the change in one place, as opposed to all three sizes.

Example: if you're just using the `outline` variant: https://codesandbox.io/s/demo-customise-input-borderradius-ucghd
